From 355a24eef73970f97bf2c71a89883139b40ec9d8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 29 Oct 2020 14:03:44 +0200
Subject: [PATCH] [SW-17915] gaudi: remove pcie_en bit strap toggle

Since the very large grace period is over and this functionality
prevents us to implement the new reset sequence and apply security
settings, we need to remove the code toggling the PCIE_EN bit in the
straps register.
Remove it for good.

Change-Id: I46de4df79439770046bcdbcd73854463459fe8d1
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 9f3996f..cae6495 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -171,12 +171,7 @@
  */
 static void preboot_pcie_enable(void)
 {
-	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
 	u32 reset_cause;
-
-	/* Check if already done by BTL */
-	if (boot_strap & 0x2)
-		return;
 
 	/* During hard reset triggered by LKD keep PCIe intact */
 	reset_cause = preboot_clr_flops_reset_cause();
